health-services/resource-estimation-service/src/main/java/org/egov/processor/service/ShapeFileParser.java

```diff
@@ -0,0 +1,146 @@
+     * defined in the plan configuration, and uploads the updated GeoJSON file to the file store.
+     *
+     * @param planConfigurationRequest  The plan configuration containing mapping and operation details.
+     * @param fileStoreId The file store ID of the Shapefile to be converted and parsed.
+     * @return The file store ID of the uploaded updated file, or null if an error occurred.
+     */
+    @Override
+    public Object parseFileData(PlanConfigurationRequest planConfigurationRequest, String fileStoreId, Object campaignResponse) {
+    	PlanConfiguration planConfig = planConfigurationRequest.getPlanConfiguration();
+        File geojsonFile = convertShapefileToGeoJson(planConfig, fileStoreId);
+        String geoJSONString = parsingUtil.convertFileToJsonString(geojsonFile);
+        JsonNode jsonNode = parsingUtil.parseJson(geoJSONString, objectMapper);
+
+        List<String> columnNamesList = parsingUtil.fetchAttributeNamesFromJson(jsonNode);
+        parsingUtil.validateColumnNames(columnNamesList, planConfig, fileStoreId);
+
+        Map<String, BigDecimal> resultMap = new HashMap<>();
+        Map<String, String> mappedValues = planConfig.getResourceMapping().stream()
+        		.filter(f-> f.getFilestoreId().equals(fileStoreId))
+        		.collect(Collectors.toMap(ResourceMapping::getMappedTo, ResourceMapping::getMappedFrom));
+        Map<String, BigDecimal> assumptionValueMap = calculationUtil.convertAssumptionsToMap(planConfig.getAssumptions());
+
+        calculationUtil.calculateResources(jsonNode, planConfigurationRequest, resultMap, mappedValues, assumptionValueMap);
+
+        File updatedGeojsonFile = parsingUtil.writeToFile(jsonNode, objectMapper);
+
+        return filestoreUtil.uploadFile(updatedGeojsonFile, planConfig.getTenantId());
+    }
+
+    /**
+     * Converts a Shapefile to GeoJSON format and writes it to a GeoJSON file.
+     *
+     * @param planConfig  The plan configuration containing mapping details.
+     * @param fileStoreId The file store ID of the Shapefile to be converted.
+     * @return The GeoJSON file containing the converted data.
+     */
+    public File convertShapefileToGeoJson(PlanConfiguration planConfig, String fileStoreId) {
+        File shapefile = null;
+        try {
+            shapefile = parsingUtil.extractShapeFilesFromZip(planConfig, fileStoreId, "shapefile");
+        } catch (IOException exception) {
+            log.error(exception.getMessage());
+        }
+
+        File geojsonFile = new File("geojsonfile.geojson");
+
+        SimpleFeatureSource featureSource = null;
+        DataStore dataStore;
+        try {
+            dataStore = getDataStore(shapefile);
+            String typeName = dataStore.getTypeNames()[0];
+            featureSource = dataStore.getFeatureSource(typeName);
+
+            writeFeaturesToGeoJson(featureSource, geojsonFile);
+        } catch (IOException e) {
+            throw new CustomException("CONVERSION_ERROR",e.getMessage());
```

Comment:
These errors needs to specific with code 

throw new CustomException("MODULE_FLOW_CLASSNAME_SPECIFICERROR",e.getMessage()); 

The error message also should contain action point on how to fix the issue